src/scripts/lightgbm_python/train_sweep_spec.yaml

```diff
@@ -0,0 +1,77 @@
+$schema: http://azureml/sdk-1-5/SweepComponent.json
+name: lightgbm_python_train_sweep
+version: 1.0.0
+display_name: LightGBM Sweep
+type: SweepComponent
+description: A sweep component for training hyper parameters
+is_deterministic: false
+
+#outputs:
+#  metrics:
+#    type: metrics
+#    description: metrics data
+#  saved_model:
+#    type: path
+#    description: path of the saved_model of trial run
+#  training_stats:
+#    type: path
+#    description: writes some stats file of the trial component.
+
+trial: file:./train_spec.yaml
+
+algorithm: random
+
+search_space:
+  num_iterations:
+    type: choice
+    values: [50]
+  num_leaves:
+    type: choice
+    values: [30]
+  min_data_in_leaf:
+    type: choice
+    values: [20] # default value from lightgbm docs https://testlightgbm.readthedocs.io/en/latest/Parameters.html#learning-control-parameters
+  learning_rate:
+    type: choice
+    values: [0.1] # default value from lightgbm docs https://testlightgbm.readthedocs.io/en/latest/Parameters.html#learning-control-parameters
+  max_bin:
+    type: choice
+    values: [255] # default value from lightgbm docs https://testlightgbm.readthedocs.io/en/latest/Parameters.html#learning-control-parameters
+  feature_fraction:
+    type: choice
+    values: [1.0] # default value from lightgbm docs https://testlightgbm.readthedocs.io/en/latest/Parameters.html#learning-control-parameters
+
+objective:
+  primary_metric:
+    default: valid_0.rmse
+    enum:
+      # NOTE: we have to declare all the possible metrics names here
```

Comment:
This seems suboptimal. Maybe something to tackle later.

Reply:
Sweep component specs need to pre-declare their list of metrics, so I'm using what's available, but we could ask for a change here.